openpgp/symmetric/symmetric.go

```diff
@@ -0,0 +1,166 @@
+package symmetric
+
+import (
+	"io"
+	"crypto"
+	"crypto/sha256"
+	"crypto/hmac"
+
+	"github.com/ProtonMail/go-crypto/openpgp/internal/algorithm"
+)
+
+type PublicKeyAEAD struct {
```

Comment:
This is very nitpicky but this should probably be `AEADPublicKey`, to keep some symmetry with `ecdh.PublicKey`, etc (and similarly for the other classes)